Fix memory leak in Find in Files

Klesomik opened the pull request:
I have implemented a fix for memory leak by hint of @Zylann in issue #31570

_Bugsquad edit: Fix #31570_

Zylann commented on editor/find_in_files.cpp:
```diff
@@ -221,7 +221,7 @@ float FindInFiles::get_progress() const {
 void FindInFiles::_scan_dir(String path, PoolStringArray &out_folders) {
 
-	DirAccess *dir = DirAccess::open(path);
+	DirAccessRef *dir = DirAccessRef::open(path);
```
`DirAccessRef` is a wrapper to a pointer, there is no need to use a pointer type here

Zylann commented on editor/find_in_files.cpp:
```diff
@@ -221,16 +221,16 @@ float FindInFiles::get_progress() const {
 	}
 
-	dir->list_dir_begin();
+	dir.list_dir_begin();
```
This won't compile either. It has to remain `->`. DirAccessRef is a wrapper to pointer, you don't need to change this. Please try to compile Godot to be sure your change works.

akien-mga commented on editor/find_in_files.cpp:
```diff
@@ -221,8 +221,8 @@ float FindInFiles::get_progress() const {
-	DirAccess *dir = DirAccess::open(path);
-	if (dir == NULL) {
+	DirAccessRef dir = DirAccess::open(path);
+	if (dir) {
```
That's wrong, `if (dir)` is like `if (dir != NULL)`, which is the opposite of the previous code. That will break the editor.

akien-mga replied:
Literally the only change needed here was to replace `DirAccess *` by `DirAccessRef`, which is equivalent but freed when out of scope, or add a `memdelete(dir)`  before any return.

akien-mga commented:
Eventually you should squash the commits together into one, and edit the resulting commit log to be more explicit about what the commit does (fixing a memory leak). Mentioning the GitHub issue being fixed is good, but it's not sufficient when reading the changelog (`git log`).

See https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#format-your-commit-logs-with-readability-in-mind

Commit:
Memory leak fixed

In some functions in editor/find_in_files.cpp was detected a memory leak
DirAccess* and FileAccess* were replaced by DirAccessRef and FileAccessRef
DirAccessRef and FileAccessRef are just wrappers for DirAccess* and FileAccess*

Fixes for issue #31659

akien-mga commented:
Thanks!

Klesomik commented:
Thank you so much! 
I've learned a lot of new things by your replies. 
Your community is really firendly to newbies!
